// strawpoll-bots/build/lib/getData.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const node_fetch_1 = require("node-fetch");
function getData(url, agent, protection) {
    return __awaiter(this, void 0, void 0, function* () {
        const req = yield node_fetch_1.default(url, {
            method: 'GET',
            agent: protection ? agent : null
        });
        const res = yield req.text();
        const data = {
            secToken: res.match(/\w+="security-token".*value="(\w+)"/i)[1],
            authToken: res.match(/\w+="field-authenticity-token".*name="(\w+)"/i)[1],
            options: res.match(/\w+="options"\s*value="(\d+)"/g).map(opt => opt.match(/value="(\w+)"/i)[1])
        };
        return data;
    });
}
exports.default = getData;
